Google.PowerShell/Container/GkeNodePoolCmdlets.cs

```diff
@@ -626,15 +628,124 @@ protected override void ProcessRecord()
             return getRequest.Execute();
         }
     }
+
+    /// <summary>
+    /// <para type="synopsis">
+    /// Removes a Google Container NodePool from a Cluster.
+    /// </para>
+    /// <para type="description">
+    /// Removes a Google Container NodePool from a Cluster.
+    /// If -Project and -Zone are not specified, the cmdlets will default
+    /// to the default project and zone.
+    /// </para>
+    /// <example>
+    ///   <code>
+    ///   PS C:\> Remove-GkeCluster -ClusterName "my-cluster" `
+    ///                             -Zone "us-west1-b" `
+    ///                             -NodePoolName "my-nodepool"
+    ///   </code>
+    ///   <para>
+    ///   Removes the node pool "my-nodepool" in cluster "my-cluster" in the zone
+    ///   "us-west1-b" of the default project.
+    ///   </para>
+    /// </example>
+    /// <para type="link" uri="(https://cloud.google.com/container-engine/docs/node-pools)">
+    /// [Node Pools]
+    /// </para>
+    /// <para type="link" uri="(https://cloud.google.com/container-engine/docs/clusters/)">
+    /// [Container Clusters]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, "GkeNodePool", SupportsShouldProcess = true)]
+    public class RemoveGkeNodePoolCmdlet : GkeCmdlet
+    {
+        /// <summary>
+        /// <para type="description">
+        /// The project that the container node pool belong to.
+        /// This parameter defaults to the project in the Cloud SDK config.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        public override string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The zone that the container node pool belong to.
+        /// This parameter defaults to the project in the Cloud SDK config.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Zone)]
+        public string Zone { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The name of the node pool to be removed.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, Position = 0)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(NodePool),
+            Property = nameof(Google.Apis.Container.v1.Data.NodePool.Name))]
+        [ValidateNotNullOrEmpty]
+        public string NodePoolName { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The name of the container cluster that the node pool is in.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, Position = 1)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Google.Apis.Container.v1.Data.Cluster),
+            Property = nameof(Google.Apis.Container.v1.Data.Cluster.Name))]
```

Comment:
I had not realized this earlier, but `Google.Apis.Container.v1.Data.Cluster` has a `Zone` property, and there for needs a separate parameter and parameter set (in all cmdlets that might take a `Cluster` object.
Oddly it does not have a project property. Can we pull that from the self-link, or will we have to supply it always as a separate parameter?